src/tracking.c

```diff
@@ -58,6 +58,12 @@ typedef struct bcastState {
                        prefix. */
 } bcastState;
 
+/* This is used to store an invalidation key pending to flush */
+typedef struct invalidatedKey {
+    uint64_t client_id;
```

Comment:
@oranagra Another question is if we use `current_client`, whether we need not store `client_id`. And we maybe don't need `trackingHandlePendingKeyInvalidations` in `beforeSleep`.

Reply:
you mean we rely on `afterCommand` (and `processCommand`) to always do the work?
it sounds possible, but then we'll want an assert in beforeSleep to make sure the list is empty.

i think the only thing that worries me is module operations done from inside a timer callback.
but since the current_client will be NULL, we won't append to our list.
let's git it a try..

Reply:
Yes. Let me do it.

Reply:
When doing this, I think about a potential bug.
We only send invalidation messages when we are not in nested call.
If `current_client` of nested `call` and parent `call` are not same and we don't record `client`, we will send invalidation messages wrong.
Currently it is impossible as blocked operation is denied in nested call.

Reply:
right, blocked clients will be handled later in beforeSleep, never inside the nested call.
but i think it makes sense that whenever we decide to postpone an invalidation, we'll make sure current_client is non-null, and we'll do that also when we flush the postponed list (in afterCommand and alike).
if we're worried of other similar issues, let's figure out the assertions to validate them.

Reply:
I mean we only store `key` but not `client_id`. In nested call we postpone an invalidation. Next time when this invalidation is handled, `current_client` is no longer the same client, so we don't know which client this invalidation for.

Reply:
ok, in that case there's nothing to assert for..  we just need to make sure not to run commands on behalf of other clients inside nested call of one client (and if we do, at least set current_client correctly).
bottom line i think we can ignore this and proceed.

Reply:
ohh, actually you mean that setting the current_client will hide the assertion...
well anyway, i still think it's safe to ignore that, it just means that when we change current_client we need to flush the queued invalidations.
maybe we can add wrapper methods that set current_client and avoid setting it directly in the code. but i don't think that's necessary now.